api/src/convert/tx_out.rs

```diff
@@ -117,14 +121,14 @@ mod tests {
             value: 1u64 << 13,
             token_id: Mob::ID,
         };
-        let source = tx::TxOut {
-            masked_amount: MaskedAmount::new(amount, &RistrettoPublic::from_random(&mut rng))
-                .unwrap(),
-            target_key: RistrettoPublic::from_random(&mut rng).into(),
-            public_key: RistrettoPublic::from_random(&mut rng).into(),
-            e_fog_hint: (&[0u8; ENCRYPTED_FOG_HINT_LEN]).into(),
-            e_memo: Some((*GenericArray::from_slice(&[9u8; 66])).into()),
-        };
+        let source = tx::TxOut::new(
+            BlockVersion::MAX,
+            amount,
+            &PublicAddress::from_random(&mut rng),
```

Comment:
Is PublicAddress the right abstraction here? It's a more concise container but I think it's a confusing context because target & txout pubkey don't really match the category PublicAddress was designed for (i.e. a tuple of EC points uniquely addressing a user). 

This could be re-addressed to have the documentation of PublicAddress state that it can be used generalize to related pairs? Or make a separate struct like TxOutPublicAddress?